config/map_fetcher.go

```diff
@@ -0,0 +1,28 @@
+package config
+
+import "strconv"
+
+// mapFetcher provides an implementation of the Fetcher interface by wrapping
+// the `map[string]string` type.
+type mapFetcher map[string]string
```

Comment:
Seeing how it's actually used as a drop-in test replacement for other types which are safe in concurrent use I think it needs its own mutex.